api/app/weather_models/precip_model.py

```diff
@@ -0,0 +1,46 @@
+import logging
+import math
+from datetime import datetime
+from typing import List
+from app.weather_models.linear_model import LinearModel
+from app.weather_models.utils import construct_dictionary_from_list_by_property
+
+logger = logging.getLogger(__name__)
+
+class PrecipModel():
```

Comment:
Should be an `RegressionModelProto` like `WindDirectionModel`

Reply:
The `add_sample` function of this class requires a unique implementation and does not accept the `HourlyActual` and `ModelRunPrediction` parameters required by the function signature in `RegressionModelProto`. I could just remove the `add_sample` abstract function from `RegressionModelProto`?

Reply:
Yeah, I like the latter (removing `add_sample` from `RegressionModelProto`), I think this interface will evolve as we see what the different weather parameters need.

Reply:
Oh that may not work since there's a few places that expect to be calling `add_sample` on a model -- feel free to disregard this comment, precip is way different than others.